@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1581,8 +1586,8 @@ export class TextElementsRenderer {
                             this.m_viewState.frameNumber,
                             renderParams.time
                         );
-                        if (textRenderState.isVisible()) {
-                            textRenderState.startFadeOut(
+                        if (hasText && textRenderState!.isVisible()) {
+                            textRenderState!.startFadeOut(
```

Comment:
should this be moved into the if(dorendertext)? I don't know why this starts to fade out here, and not closer to where the text is processed.

Reply:
this function is in general a mess. I've refactored it in a later commit trying to simplify it as much as possible. Let's better have a deeper review on that commit, I'll add you to it.